### PR TITLE
Fix unnecessary panel reload and refactor top-level host and inventory routing components

### LINKED ISSUE #6191: (WFJT | maybe others) tab switching reloads entire panel
##### ISSUE TYPE
 - Bug Report

##### SUMMARY
We're not loading / unloading the correct set of components when switching between tabs on some of our views. Best illustrated by comparison - see gifs below.

Regular JT:
template_tab_load

WFJT:
wfjt_tab_load

### PR DESCRIPTION
**Pull request by jakemcdermott**
##### SUMMARY
The primary goal is to fix https://github.com/ansible/awx/issues/6191#issuecomment-596530283

While there, I did some light refactoring:
* Remove route render props
* Remove unneeded load checks
* Make PageSection component placement consistent with other screens

### CODE REVIEW

**Comment by jakemcdermott** [comment]
recheck

File: awx/ui_next/src/screens/Inventory/Inventory.jsx [hunk 1]
```diff
@@ -111,67 +93,56 @@ function Inventory({ i18n, setBreadcrumb }) {
   return (
     <PageSection>
       <Card>
-        {cardHeader}
+        {['edit', 'add', 'groups/', 'hosts/'].some(name =>
```
Comment (AlexSCorey) [inline awx/ui_next/src/screens/Inventory/Inventory.jsx:96]: I like this pattern.

**Review by AlexSCorey — APPROVED** [review]
LGTM

**Comment by jakemcdermott** [comment]
regate

**Pull request merged**